config.go

```diff
@@ -1,46 +1,61 @@
 package main
 
 import (
-	"github.com/devangel/config"
 	"os"
 	"path/filepath"
-	"strings"
-	//"fmt"
+
+	"github.com/devangel/config"
 )
 
 var Config = config.NewConfig("force")
 
-func GetSourceDir() (src string, err error) {
-	// Last element is default
-	var sourceDirs = []string{
-		"src",
-		"metadata",
+var sourceDirs = []string{
+	"src",
+	"metadata",
+}
+
+// IsSourceDir returns a boolean indicating that dir is actually a Salesforce
+// source directory.
+func IsSourceDir(dir string) bool {
```

Comment:
I think what you have got looks good if it is ignoring the "package"
package.xml in determining the root/source dir.  I'll merge the PR.

D

On Fri, Jul 29, 2016 at 3:26 PM, Simon Law <notifications@github.com> wrote:

> In config.go
> <https://github.com/heroku/force/pull/294#discussion_r72869652>:
>
> >  )
> >
> >  var Config = config.NewConfig("force")
> >
> > -func GetSourceDir() (src string, err error) {
> > -	// Last element is default
> > -	var sourceDirs = []string{
> > -		"src",
> > -		"metadata",
> > +var sourceDirs = []string{
> > +	"src",
> > +	"metadata",
> > +}
> > +
> > +// IsSourceDir returns a boolean indicating that dir is actually a Salesforce
> > +// source directory.
> > +func IsSourceDir(dir string) bool {
>
> Just like the previous code, this gives precedence for the src directory
> before the metadata directory. Unlike the previous code, it doesn't look
> too far up the parent tree to determine if it has found it.
>
> As for the Package folder, what is the semantic for that? Are you saying
> that we should also match Package/package.xml? Or ignore it? Because this
> code should ignore it, although it won't ignore Package/src/package.xml
> or Package/metadata/package.xml.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/heroku/force/pull/294/files/188d84d4bc82c2a176c58999da9512e5c64deb6f#r72869652>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAHGHgGH0XleybNGOpH7Q0LchUvaTU-jks5qan38gaJpZM4IttBn>
> .
>



-- 
Thanks,
Dave